### DISCUSSION TITLE
detect code smells for iOS

### DISCUSSION THREAD

**Discussion by xijun** [green-code-initiative/creedengo-android-java#65] [Q&A]
Hello, I have tried to use ecoMobile on the iOS test project. It seems it didn't detect any code smell implemented by ecoMobile built-in rule.
When I run ecoMobile docker run command, the logs say that the iOS sonar plugin docker image is null.
Am I doing something wrong ?

**Comment by dedece35** [comment]
Hi @xijun,

please give use a snapshot or a part of your logs including this error.

thank you.

**Reply by xijun** [reply]
Hello, thanks for answering. This is what I have when I run the docker command to launch sonarqube locally.

<img width="1349" alt="Screenshot 2023-04-24 at 10 15 47" src="https://user-images.githubusercontent.com/3062607/233938661-09a93d36-4e1f-4a1d-a035-b987e4d3d469.png">

I don't have any errors when scanning. But on SonarQube I just see issues that I would be able to see if I ran SwiftLint

**Reply by zippy1978** [reply]
Hi @xijun ,

I think I get the problem: you probably did not enabled ecoCode rules in you quality profile.

When you install the plugin (along with the Swift plugin), you get 2 profiles like this : 

![Capture d’écran 2023-05-05 à 01 02 37](https://user-images.githubusercontent.com/885183/236348285-0238551c-f393-4f71-b400-7e54b6806d40.png)

But you'll see as well that the "ecoCode" profile is not set as default for Swift.
You need to explicitly create a custom profile (mixing the 2 built-in one for example), and set it as default for all your Swift projects.

Here is an example, where I created a new profile by extending the Swift one with the ecoCode one (and set it as default):

![Capture d’écran 2023-05-05 à 01 07 55](https://user-images.githubusercontent.com/885183/236348772-1c05ca0d-38c0-4d9e-b7a1-4f9f120f85a0.png)
